### PR TITLE
Revert "warn newer available version of the x tool"

### PR DESCRIPTION
**Pull request by DebugSteven**
Reverts rust-lang/rust#104552

Running the x executable directly created an [issue](https://github.com/rust-lang/rust/issues/106469) here. There are other options for warning a user that a newer version of x exists in the issue's discussion as well.

r? @jyn514

### CODE REVIEW

File: src/tools/x/src/main.rs [hunk 1]
```diff
@@ -52,14 +52,6 @@ fn exec_or_status(command: &mut Command) -> io::Result<ExitStatus> {
...
 
 fn main() {
-    match env::args().skip(1).next().as_deref() {
-        Some("--wrapper-version") => {
-            let version = env!("CARGO_PKG_VERSION");
-            println!("{}", version);
-            return;
-        }
-        _ => {}
-    }
```
Comment (jyn514) [inline src/tools/x/src/main.rs:62]: let's keep this so we have fewer people getting warnings when we eventually reland this change.

**Review by jyn514 — COMMENTED** [review]
thanks! LGTM other than the nit

**Comment by pnkfelix** [comment]
Or, oh, wait, @jyn514 delegated to the author. (I misinterpreted the earlier interaction with bors)

@rustbot label: -S-waiting-on-review +S-waiting-on-author

**Pull request merged**